src/ixdat/db.py

```diff
@@ -197,33 +230,63 @@ def set_backend(self, backend):
         """Backends set obj.backend here after loading/saving a Saveable obj"""
         self._backend = backend
 
-    def get_main_dict(self):
-        """Return dict: serializition only of the row of the object's main table"""
+    def get_main_dict(self, exclude=None):
+        """Return dict: serializition only of the row of the object's main table
+
+        Args:
+            exclude (list): List of attribute names to leave out of the dict
+        """
+        exclude = exclude or []
         if self.column_attrs is None:
             raise DataBaseError(
                 f"{self} can't be serialized because the class "
                 f"{self.__class__.__name__} hasn't defined column_attrs"
             )
-        self_as_dict = {attr: getattr(self, attr) for attr in self.column_attrs}
+        self_as_dict = {  # FIXME: probably better as loop, fix with table definitions.
+            attr: getattr(self, attr)
+            for attr in self.column_attrs
+            if attr not in exclude
+        }
         return self_as_dict
 
-    def as_dict(self):
+    def as_dict(self, exclude=None):
         """Return dict: serialization of the object main and auxiliary tables"""
-        self_as_dict = self.get_main_dict()
+
+        # So that a new object initiated constructed using the dictionary returned here
+        #   can find objects referenced by identities (i.e. s_ids for series_list),
+        #   we need to make sure they are saved in a real backend. Before adding the
+        #   id's to a list.
+        # FIXME: There would be a more precise and elegant way to do this if the id's
+        #   and corresponding attributes could be connected through class attributes.
+        #   To do with table definitions.
+        if self.child_attrs:
+            for child_object_list_name in self.child_attrs:
+                for child_obj in getattr(self, child_object_list_name):
+                    if child_obj.backend in ("none", database_backends["none"]):
```

Comment:
This is not really specific to this code, but more a thought that has occured to me a few times while reading this class. Is there a good reason for allowing the type duality in `.backend`, where it is sometimes a string, sometimes a backend and sometimes a backend class? Because it does complicate quite a few places. One might think that a backend instance and None would be sufficient, but I may not have the full overview.

Reply:
Okay I moved the string treatment to `Savable.__init__`, so backend can only be `None` or a subclass of `BackendBase` elsewhere.

Reply:
Ah yes, I see that it conflicts with my comment from earlier about not instantiating all backends up front.